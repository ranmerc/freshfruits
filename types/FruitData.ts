export default interface FruitData {
  id: number;
  name: string;
  images: string[];
  description: string;
  shelfLife: number;
  storageInstructions: string;
  countryOfOrigin: string;
}
