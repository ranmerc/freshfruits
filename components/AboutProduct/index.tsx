import FruitData from "@/types/FruitData";
import { Box, Stack, Typography } from "@mui/material";

export default function AboutProduct({ fruitData }: { fruitData: FruitData }) {
  return (
    <Box>
      <Typography variant="h5" component="h2" fontWeight={"bold"} mb={1.5}>
        Fruit Details
      </Typography>
      <Stack spacing={1}>
        <Box>
          <Typography variant="body1" component="h3" fontWeight={"bold"}>
            Description
          </Typography>
          <Typography variant="body2">{fruitData.description}</Typography>
        </Box>
        <Box>
          <Typography variant="body1" component="h3" fontWeight={"bold"}>
            Shelf Life
          </Typography>
          <Typography variant="body2">{fruitData.shelfLife} days</Typography>
        </Box>
        <Box>
          <Typography variant="body1" component="h3" fontWeight={"bold"}>
            Storage Instructions
          </Typography>
          <Typography variant="body2">
            {fruitData.storageInstructions}
          </Typography>
        </Box>
        <Box>
          <Typography variant="body1" component="h3" fontWeight={"bold"}>
            Country of Origin
          </Typography>
          <Typography variant="body2">{fruitData.countryOfOrigin}</Typography>
        </Box>
      </Stack>
    </Box>
  );
}
