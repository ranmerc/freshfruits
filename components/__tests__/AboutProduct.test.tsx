import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import AboutProduct from "../AboutProduct";
import FruitData from "@/types/FruitData";

const fruitData: FruitData = {
  id: 1,
  name: "Fruit",
  countryOfOrigin: "India",
  description: "This is a delicious fruit",
  images: ["/fruit-1.jpeg", "/fruit-2.jpeg"],
  shelfLife: 4,
  storageInstructions: "Store in cool dry place",
};

describe("About Product Component", () => {
  it("Renders heading", () => {
    render(<AboutProduct fruitData={fruitData} />);

    expect(
      screen.getByRole("heading", { level: 2, name: "Fruit Details" })
    ).toBeInTheDocument();
  });

  it("Renders description", () => {
    render(<AboutProduct fruitData={fruitData} />);

    expect(
      screen.getByRole("heading", { level: 3, name: "Description" })
    ).toBeInTheDocument();

    expect(screen.getByText(fruitData.description)).toBeInTheDocument();
  });

  it("Renders shelf life", () => {
    render(<AboutProduct fruitData={fruitData} />);

    expect(
      screen.getByRole("heading", { level: 3, name: "Shelf Life" })
    ).toBeInTheDocument();

    expect(screen.getByText(`${fruitData.shelfLife} days`)).toBeInTheDocument();
  });

  it("Renders storage instructions", () => {
    render(<AboutProduct fruitData={fruitData} />);

    expect(
      screen.getByRole("heading", { level: 3, name: "Storage Instructions" })
    ).toBeInTheDocument();

    expect(screen.getByText(fruitData.storageInstructions)).toBeInTheDocument();
  });

  it("Renders country of origin", () => {
    render(<AboutProduct fruitData={fruitData} />);

    expect(
      screen.getByRole("heading", { level: 3, name: "Country of Origin" })
    ).toBeInTheDocument();

    expect(screen.getByText(fruitData.countryOfOrigin)).toBeInTheDocument();
  });
});
