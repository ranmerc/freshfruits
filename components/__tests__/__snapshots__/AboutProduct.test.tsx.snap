// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Product Component Renders correctly 1`] = `
<div
  className="MuiBox-root css-0"
>
  <h2
    className="MuiTypography-root MuiTypography-h5 css-79feuo-MuiTypography-root"
  >
    Fruit Details
  </h2>
  <div
    className="MuiStack-root css-jfdv4h-MuiStack-root"
  >
    <div
      className="MuiBox-root css-0"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        Description
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        This is a delicious fruit
      </p>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        Shelf Life
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        4
         days
      </p>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        Storage Instructions
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Store in cool dry place
      </p>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <h3
        className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      >
        Country of Origin
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        India
      </p>
    </div>
  </div>
</div>
`;
